simde/x86/gfni.h

```diff
@@ -0,0 +1,765 @@
+    for (size_t i = 0 ; i < (sizeof(r_.u8) / sizeof(r_.u8[0])) ; i++) {
+      r_.u8[i] = ((k >> i) & 1) ? r_.u8[i] : src_.u8[i];
+    }
+
+    return simde__m128i_from_private(r_);
+  }
+  #if defined(SIMDE_GFNI_ENABLE_NATIVE_ALIASES)
+    #define _mm_mask_gf2p8affineinv_epi64_epi8(src, k, x, A, b) simde_mm_mask_gf2p8affineinv_epi64_epi8(src, k, x, A, b)
+  #endif
+#endif
+
+#if defined(SIMDE_GFNI_NATIVE) && defined(SIMDE_AVX512VL_NATIVE)
+  #define simde_mm256_mask_gf2p8affineinv_epi64_epi8(src, k, x, A, b) _mm256_mask_gf2p8affineinv_epi64_epi8(src, k, x, A, b)
+#else
+  SIMDE__FUNCTION_ATTRIBUTES
+  simde__m256i
+  simde_mm256_mask_gf2p8affineinv_epi64_epi8 (simde__m256i src, simde__mmask32 k, simde__m256i x, simde__m256i A, int b) {
+    simde__m256i_private
+      src_ = simde__m256i_to_private(src);
+
+    simde__m256i r = simde_mm256_gf2p8affineinv_epi64_epi8(x, A, b);
+    simde__m256i_private r_ = simde__m256i_to_private(r);
+
+    SIMDE__VECTORIZE
+    for (size_t i = 0 ; i < (sizeof(r_.u8) / sizeof(r_.u8[0])) ; i++) {
+      r_.u8[i] = ((k >> i) & 1) ? r_.u8[i] : src_.u8[i];
+    }
+
+    return simde__m256i_from_private(r_);
+  }
+  #if defined(SIMDE_GFNI_ENABLE_NATIVE_ALIASES)
+    #define _mm256_mask_gf2p8affineinv_epi64_epi8(src, k, x, A, b) simde_mm256_mask_gf2p8affineinv_epi64_epi8(src, k, x, A, b)
+  #endif
+#endif
+
+#if defined(SIMDE_GFNI_NATIVE) && defined(SIMDE_AVX512F_NATIVE)
+  #define simde_mm512_mask_gf2p8affineinv_epi64_epi8(src, k, x, A, b) _mm512_mask_gf2p8affineinv_epi64_epi8(src, k, x, A, b)
+#else
+  SIMDE__FUNCTION_ATTRIBUTES
+  simde__m512i
+  simde_mm512_mask_gf2p8affineinv_epi64_epi8 (simde__m512i src, simde__mmask64 k, simde__m512i x, simde__m512i A, int b) {
+    simde__m512i_private
+      src_ = simde__m512i_to_private(src);
+
+    simde__m512i r = simde_mm512_gf2p8affineinv_epi64_epi8(x, A, b);
+    simde__m512i_private r_ = simde__m512i_to_private(r);
+
+    SIMDE__VECTORIZE
+    for (size_t i = 0 ; i < (sizeof(r_.u8) / sizeof(r_.u8[0])) ; i++) {
+      r_.u8[i] = ((k >> i) & 1) ? r_.u8[i] : src_.u8[i];
+    }
+
+    return simde__m512i_from_private(r_);
+  }
+  #if defined(SIMDE_GFNI_ENABLE_NATIVE_ALIASES)
+    #define _mm512_mask_gf2p8affineinv_epi64_epi8(src, k, x, A, b) simde_mm512_mask_gf2p8affineinv_epi64_epi8(src, k, x, A, b)
+  #endif
+#endif
+
+#if defined(SIMDE_GFNI_NATIVE) && defined(SIMDE_AVX512VL_NATIVE)
+  #define simde_mm_maskz_gf2p8affineinv_epi64_epi8(k, x, A, b) _mm_maskz_gf2p8affineinv_epi64_epi8(k, x, A, b)
+#else
+  SIMDE__FUNCTION_ATTRIBUTES
+  simde__m128i
+  simde_mm_maskz_gf2p8affineinv_epi64_epi8 (simde__mmask16 k, simde__m128i x, simde__m128i A, int b) {
+    simde__m128i r = simde_mm_gf2p8affineinv_epi64_epi8(x, A, b);
+    simde__m128i_private r_ = simde__m128i_to_private(r);
+
+    SIMDE__VECTORIZE
+    for (size_t i = 0 ; i < (sizeof(r_.u8) / sizeof(r_.u8[0])) ; i++) {
+      r_.u8[i] = ((k >> i) & 1) ? r_.u8[i] : 0;
+    }
+
+    return simde__m128i_from_private(r_);
+  }
```

Comment:
Like the mask functions, only with `simde_mm_maskz_mov_epi8` instead of `simde_mm_mask_mov_epi8`.

Reply:
OK

Reply:
Thanks again for doing this; in the future I'll try to take care of stuff like this myself…

Your changes all look good, though!